File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        ht = {}

        for i, n in enumerate(nums):
            diff = target - n
            if diff in ht:
                return [ht[diff], i]
            ht[n] = i

                

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.java
class Solution {
    // T = O(n^2)
    public int[] twoSum(int[] numbers, int target) {
        for (int i = 0; i < numbers.length; i++) {
            for (int j = 0; j < numbers.length; j++) {
                if (i != j && numbers[i] + numbers[j] == target) {
                    int smaller = i < j ? i : j;
                    int bigger = i > j ? i : j;
                    return new int[]{smaller + 1, bigger + 1};
                }
            }
        }

        return new int[2];
    }
}
